test/tls_test.go

```diff
@@ -859,6 +859,55 @@ func TestTLSAuthorizationShortTimeout(t *testing.T) {
 	}
 }
 
+func TestClientTLSAndNonTLSConnections(t *testing.T) {
+	s, opts := RunServerWithConfig("./configs/tls_mixed.conf")
```

Comment:
You could create the config on the fly and maybe have a t.Run() with the 2 modes, with and without client cert verification.

Reply:
For what purpose? TLS client HELLO proto always the same.